content/into.md

```diff
@@ -0,0 +1,306 @@
+
+`into` is defined as follows in the companion object of the `scala.Conversion` class:
+```scala
+opaque type into[T] >: T = T
+```
+Types of the form `into[T]` are treated specially during type checking. If the expected type of an expression is `into[T]` then an implicit conversion to that type can be inserted without the need for a language import.
+
+Note: Unlike other types, `into` starts with a lower-case letter. This emphasizes the fact that `into` is treated specially by the compiler, by making `into` look more like a keyword than a regular type.
+
+### Example 1
+
+```scala
+given Conversion[Array[Int], IterableOnce[Int]] = wrapIntArray
+val xs: List[Int] = List(1)
+val ys: Array[Int] = Array(2, 3)
+xs ++ ys
+```
+This inserts the given conversion on the `ys` argument in `xs ++ ys`. It typechecks without a feature warning since the formal parameter of `++` is of type `into[IterableOnce]`, which is also the expected type of `ys`.
+
+### Example 2
+
+Consider a simple expression AST type:
+```scala
+enum Expr:
+  case Neg(e: Expr)
+  case Add(e1: Expr, e2: Expr)
+  case Const(n: Int)
+import Expr.*
+```
+Say we'd like to build `Expr` trees without explicit `Const` wrapping, as in `Add(1, Neg(2))`. The usual way to achieve this is with an implicit conversion from `Int` to `Const`:
+```scala
+given Conversion[Int, Const] = Const(_)
+```
+Normally, that would require a language import in all source modules that construct `Expr` trees. We can avoid this requirement on user code by declaring `Neg` and `Add` with `into` parameters:
+```scala
+enum Expr:
+  case Neg(e: into[Expr])
+  case Add(e1: into[Expr], e2: into[Expr])
+  case Const(n: Int)
+```
+This would allow conversions from `Int` to `Const` when constructing trees but not elsewhere.
+
+### `into` in Function Results
+
+`into` allows conversions everywhere it appears as expected type, including in the results of function arguments. For instance, consider the new proposed signature of the `flatMap` method on `List[A]`:
+
+```scala
+  def flatMap[B](f: A => into[IterableOnce[B]]): List[B]
+```
+This accepts all actual arguments `f` that, when applied to an `A`, give a result
+that is convertible to `IterableOnce[B]`. So the following would work:
+```scala
+scala> val xs = List(1, 2, 3)
+scala> xs.flatMap(x => x.toString * x)
+val res2: List[Char] = List(1, 2, 2, 3, 3, 3)
+```
+Here, the conversion from `String` to `Iterable[Char]` is applied on the results of `flatMap`'s function argument when it is applied to the elements of `xs`.
+
+### Vararg arguments
+
+When applied to a vararg parameter, `into` allows a conversion on each argument value individually. For example, consider a method `concatAll` that concatenates a variable
+number of `IterableOnce[Char]` arguments, and also allows implicit conversions into `IterableOnce[Char]`:
+
+```scala
+def concatAll(xss: into[IterableOnce[Char]]*): List[Char] =
+  xss.foldLeft(List[Char]())(_ ++ _)
```

Comment:
This is a quadratic cost but a not uncommon kind of mistake. Can we use something as an example that isn't an anti pattern? For instance prepending with a foldRight for instance, or `xss.flatten.to(List)`